### PR TITLE
Improve autogen schema of many operators

### PR DESCRIPTION
**Pull request by Ingkarat**
RTO = relevantToOptimizer list

- AdditiveChi2Sampler - Add a constraint. Modify HPs schema and RTO.
- ARDRegression - Add a constraint. Add "regressor" tag.
- BayesianRidge - Add "regressor" tag.
- CalibratedClassifierCV - Add "classifier" tag. Modify HPs schema.
- ComplementNB - Add "classifier" tag.
- ElasticNet - Add "regressor" tag.
- ElasticNetCV - Add a constraint. Add "regressor" tag. Modify HPs schema and RTO.
- FactorAnalysis - Add 2 constraints. Modify HPs schema and RTO.
- FastICA - Add a constraint.
- GaussianProcessClassifier - Modify HPs schema.
- GaussianRandomProjection - Add a constraint.
- HuberRegressor - Modify HPs schema.
- KBinsDiscretizer - Add a constraint. Modify HPs schema and RTO.
- KernelRidge - Add "regressor" tag.
- LabelPropagation - Add "classifier" tag. Modify HPs schema.
- LabelSpreading - Add "classifier" tag. Modify HPs schema.
- Lars - Add "regressor" tag. Modify RTO.
- LarsCV - Add "regressor" tag. Modify HPs schema.
- Lasso - Add "regressor" tag.
- LassoCV - Add "regressor" tag.
- LassoLars - Add "regressor" tag.
- LassoLarsCV - Add "regressor" tag.
- LassoLarsIC - Add 2 constraints. Add "regressor" tag. Modify HPs schema.

### CODE REVIEW

**Comment by hirzel** [comment]
Hi @Ingkarat thank you so much for this PR! The static checks are currently failing due to an unrelated problem that has nothing to do with your PR. This is being actively worked on, and I will let you know when it is resolved. In the meantime, I'll review your code changes, and add comments (if any).

File: lale/lib/autogen/additive_chi2_sampler.py [hunk 1]
```diff
@@ -28,24 +28,45 @@ def transform(self, X):
...
+            "description": "From /kernel_approximation.py:AdditiveChi2Sampler:fit, Exception: raise ValueError(     'If "
+            "sample_steps is not in [1, 2, 3], you need to provide sample_interval') ",
+            "anyOf": [
+                {
+                    "type": "object",
+                    "properties": {"sample_interval": {"not": {"enum": [None]}}},
+                },
+                {"type": "object", "properties": {"sample_steps": {"enum": [1]}}},
+                {"type": "object", "properties": {"sample_steps": {"enum": [2]}}},
+                {"type": "object", "properties": {"sample_steps": {"enum": [3]}}},
```
Comment (hirzel) [inline lale/lib/autogen/additive_chi2_sampler.py:67]: Can you combine these three lines with `{"enum": [1, 2, 3]}` please?

File: lale/lib/autogen/factor_analysis.py [hunk 2]
```diff
@@ -38,17 +38,24 @@ def transform(self, X):
...
                 "iterated_power",
             ],
             "additionalProperties": False,
             "properties": {
                 "n_components": {
-                    "enum": ["int", None],
+                    "anyOf": [
+                        {
+                            "type": "integer",
+                            "minimumForOptimizer": 2,
```
Comment (hirzel) [inline lale/lib/autogen/factor_analysis.py:53]: There should probably also be a `minimum`, not just a `minimumForOptimizer`. Need to figure out what that minimum is, based on dynamic exceptions.

File: lale/lib/autogen/lasso_lars_ic.py [hunk 3]
```diff
@@ -114,6 +115,31 @@ def predict(self, X):
...
+            "anyOf": [
+                {"XXX TODO XXX": "input X is not None"},
+                {"type": "object", "properties": {"precompute": {"enum": [None]}}},
+            ],
+        },
+        {
+            "description": "From /linear_model/_least_angle.py:None:_lars_path_solver, Exception: raise ValueError('X and Gram cannot both be unspecified.') ",
+            "anyOf": [
+                {
+                    "allOf": [
```
Comment (hirzel) [inline lale/lib/autogen/lasso_lars_ic.py:129]: This entire `allOf` could be replaced by
```python
{"type": "object", "properties": {"precompute": {"not": {"enum": [None, False]}}}}
```

**Review by hirzel — APPROVED** [review]
Looks great overall! I suggested a few minor changes, see detailed comments.

**Comment by Ingkarat** [comment]
Thank you fore the review. I will make the changes.

**Pull request merged**